lib/blob/blobstore.c

```diff
@@ -3239,13 +3275,13 @@ blob_request_submit_rw_iov(struct spdk_blob *blob, struct spdk_io_channel *_chan
 
 			return;
 		}
-
 		is_allocated = blob_calculate_lba_and_lba_count(blob, offset, length, &lba, &lba_count);
 
 		if (read) {
 			spdk_bs_sequence_t *seq;
 
 			if (ext_io_flags & SPDK_NVME_IO_FLAGS_UNWRITTEN_READ_FAIL) {
+				is_allocated = blob_ancestor_calc_lba_and_lba_count(blob, offset, length, &lba, &lba_count);
```

Comment:
aren't we calculating it twice here? on 3278 and 3284?

Reply:
Yes. As the claculation is not expensive thought of keeping the checks for `ext_io_flag` for read just once.
It is kind of trade-off between keeping checks 2 times OR execute the calculation 2 times when the read & ext_io_flag condition is meet.

What is your suggestion ? We will keep the checks 2 times and execution of of the function once ?

Reply:
Yes I see, it is indeed a tradeoff.
I think it's ok as is then as the original calculation `blob_calculate_lba_and_lba_count` is cheap, it doesn't iterate?

Reply:
No. `blob_calculate_lba_and_lba_count` function and sub-functions called inside it, doesn't iterate.